Lib/site.py

```diff
@@ -372,6 +372,18 @@ def setcopyright():
 def sethelper():
     builtins.help = _sitebuiltins._Helper()
 
+
+def set_history_file():
+    """Check if PYTHONHISTORY environment variable
+     is set and define it as python_history file,
+     if not use the default ~/.python_history file.
+    """
+    history_file = os.environ.get("PYTHONHISTORY")
+    if history_file:
+        return history_file
+    return os.path.join(os.path.expanduser('~'),
+        '.python_history')
```

Comment:
You could also do something like:

```python
    return os.environ.get(
        'PYTHONHISTORY',
        os.path.expanduser(os.path.join('~', '.python_history')))
```

Also `set_history_file()` is a little bit of a misnomer since it's not actually setting it, but calculating it instead.  OTOH, I bet the more compact form could just be inlined below.